.travis.yml

```diff
@@ -5,16 +5,16 @@ os:
 language: go
 
 go:
-    - 1.8
+    - 1.9.2
 
 install: true
 
 before_install:
   - make deps
 
 script:
-  - go vet
-  - go test -v -race -coverprofile=coverage.txt -covermode=atomic .
+  - go vet ./...
+  - go test -v -race -coverprofile=coverage.txt -covermode=atomic ./...
```

Comment:
You can replace that with:
```
go list -f '{{if (len .TestGoFiles)}}{{.ImportPath}}{{end}}' ./... | grep -v /vendor/ | xargs -L1 -i bash -c 'go test -v -race -coverprofile="coverage.$(echo "$1" | md5sum | head -c 16).txt" -covermode=atomic "$1"' "" {}
```

It got a bit complex but works, I just tested.

Reply:
What is this doing?

Reply:
it lists packages with tests and then runs `go test -cover -coverprofile="$(path specific name)" $package` on all packages with tests.